analyzers/src/SonarAnalyzer.CSharp/Trackers/CSharpObjectCreationTracker.cs

```diff
@@ -21,22 +21,24 @@
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using SonarAnalyzer.Extensions;
+using SonarAnalyzer.Wrappers;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Helpers.Trackers
 {
     public class CSharpObjectCreationTracker : ObjectCreationTracker<SyntaxKind>
     {
         protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
-        protected override SyntaxKind[] TrackedSyntaxKinds { get; } = { SyntaxKind.ObjectCreationExpression };
+        protected override SyntaxKind[] TrackedSyntaxKinds { get; } = { SyntaxKind.ObjectCreationExpression, SyntaxKindEx.ImplicitObjectCreationExpression };
 
         internal override Condition ArgumentAtIndexIsConst(int index) =>
-            context => ((ObjectCreationExpressionSyntax)context.Node).ArgumentList is { } argumentList
+            context => ObjectCreationFactory.Create(context.Node).ArgumentList is { } argumentList
```

Comment:
Let's try this syntax to cover the case with `ArgumentList` being `null`
```
var x = new Something {Property=42};
```

Reply:
Had to craft a custom test for this, since this is not invocable when constructor has no parameters in the existing rules.